src/main/java/be/seeseemelk/mockbukkit/inventory/meta/MapMetaMock.java

```diff
@@ -0,0 +1,166 @@
+package be.seeseemelk.mockbukkit.inventory.meta;
+
+import org.bukkit.Color;
+import org.bukkit.inventory.meta.MapMeta;
+import org.bukkit.map.MapView;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+public class MapMetaMock extends ItemMetaMock implements MapMeta
+{
+
+	private static final byte SCALING_EMPTY = (byte) 0;
+	private static final byte SCALING_TRUE = (byte) 1;
+	private static final byte SCALING_FALSE = (byte) 2;
+
+	private Integer mapId;
+	private MapView mapView;
+	private Color color;
+	private byte scaling = SCALING_EMPTY;
+
+	public MapMetaMock()
+	{
+		super();
+	}
+
+	public MapMetaMock(MapMeta meta)
+	{
+		super(meta);
+
+		try
+		{
+			this.mapId = meta.getMapId();
+		} // If no map ID is set, it will throw a NPE when trying to convert the Integer to an int.
+		catch (NullPointerException ignored)
+		{
+		}
+		this.mapView = meta.getMapView();
+		this.color = meta.getColor();
+		this.scaling = meta instanceof MapMetaMock metaMock ? metaMock.scaling : meta.isScaling() ? SCALING_TRUE : SCALING_FALSE;
+	}
+
+	@Override
+	public boolean hasMapId()
+	{
+		return mapId != null;
+	}
+
+	@Override
+	public int getMapId()
+	{
+		return mapId;
```

Comment:
This should probably throw a proper exception instead of a NullPointer if no mapId is set.

Reply:
[That's what CraftBukkit does](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaMap.java#145-148). You're supposed to check `#hasMapId` before getting it.